File: src/main/java/org/labun/jooq/generator/util/FilePathResolver.java
package org.labun.jooq.generator.util;

import org.labun.jooq.generator.config.CodeGenerationConfig;

import java.io.File;
import java.nio.file.Path;
import java.nio.file.Paths;

/**
 * @author Konstantin Labun
 */
public class FilePathResolver {

    private CodeGenerationConfig config;

    public FilePathResolver(CodeGenerationConfig config) {
        this.config = config;
    }

    public Path resolveFilePath(String className) {
        return Paths.get(config.generatedSourcesRoot(), fileName(className));
    }

    private String fileName(String className) {
        return packagePath() + File.separator + className + ".java";
    }

    private String packagePath() {
        return packageName().replace(".", File.separator);
    }

    private String packageName() {
        return config.packageName();
    }
}

File: src/main/kotlin/org/labun/jooq/generator/util/FilePathResolver.kt
package org.labun.jooq.generator.util

import org.labun.jooq.generator.config.SubGeneratorConfig
import java.io.File
import java.nio.file.Paths

/**
 * @author Konstantin Labun
 */
class FilePathResolver(private val taskConfigSub: SubGeneratorConfig) {

  fun resolveFilePath(className: String) =
      Paths.get(taskConfigSub.generatedSourcesRoot, fileName(className))

  private fun fileName(className: String) = "${packagePath()}${File.separator}${className}.java"

  private fun packagePath() = packageName().replace(".", File.separator)

  private fun packageName() = taskConfigSub.packageName
}